nebullvm/operations/optimizations/compilers/faster_transformer/__init__.py

```diff
@@ -0,0 +1,172 @@
+from typing import List, Tuple, Union
+from copy import deepcopy
+
+from nebullvm.config import QUANTIZATION_DATA_NUM
+from nebullvm.operations.optimizations.compilers.base import Compiler
+from nebullvm.operations.optimizations.compilers.quantizations.pytorch import (
+    quantize_pytorch,
+)
+from nebullvm.operations.optimizations.compilers.quantizations.utils import (
+    check_quantization,
+)
+from nebullvm.optional_modules.torch import (
+    GraphModule,
+    Module,
+    ScriptModule,
+    symbolic_trace,
+    torch,
+)
+from nebullvm.tools.base import QuantizationType, DeviceType
+from nebullvm.tools.data import DataManager
+from nebullvm.tools.transformations import MultiStageTransformation
+from nebullvm.tools.huggingface import PyTorchTransformerWrapper
+
+from nebullvm.operations.optimizations.compilers.faster_transformer.bert import (  # noqa: E501
+    detect_and_swap_bert_model,
+)
+from nebullvm.operations.optimizations.compilers.utils import (
+    get_faster_transformer_repo_path,
+)
+
+default_lib_path = str(
+    get_faster_transformer_repo_path()
+    / "build"
+    / "lib"
+    / "libth_transformer.so"
+)
+
+
+def detect_and_swap_model(model, data_type="fp16", remove_padding=False):
+    """currently only supports:
+    - BertModel and model with BertModel as .bert attribute
+    """
+    model = detect_and_swap_bert_model(
```

Comment:
no, it's not supported here because `optimize_model` doesn't evaluate model based on `generate`, and even the `.generate` is a bit different from the original `.generate`.